File: Finance.WebAPI/finance-spa/models/user-preference.ts
export type NumberGroupingStyle = "international" | "indian"
export type FinancialYearMode = "indian" | "american" | "custom"

export type UserPreference = {
  userId: string
  numberGroupingStyle: NumberGroupingStyle
  financialYearMode: FinancialYearMode
  customFinancialYearStartDate: string | null
  updatedAt: string
}

File: Finance.Domain/Entities/Core/UserPreference.cs
namespace Finance.Domain.Entities.Core
{
    public class UserPreference
    {
        public Guid UserId { get; set; }

        public string NumberGroupingStyle { get; set; } = "international";

        public string FinancialYearMode { get; set; } = "indian";

        public DateTime? CustomFinancialYearStartDate { get; set; }

        public DateTime UpdatedAt { get; set; }

        public User User { get; set; } = null!;
    }
}
